File: backend/Entities/Concrete/Video.cs
using Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Concrete
{
    public class Video : IEntity
    {
        public int VideoID { get; set; }
        public int SectionID { get; set; }
        public int? Order { get; set; }
        public string Title { get; set; }
        public int Duration { get; set; }
        public string URL { get; set; }
    }
}

File: lib/models/video.dart
class Video {
  final int videoID;
  final int sectionID;
  final int? order;
  final String? title;
  final int? duration;
  final String? url;

  Video({
    required this.videoID,
    required this.sectionID,
    this.order,
    this.title,
    this.duration,
    this.url,
  });

  factory Video.fromMap(Map<String, dynamic> map) {
    return Video(
      videoID: map['videoID'] ?? 0,
      sectionID: map['sectionID'] ?? 0,
      order: map['order'],
      title: map['title'],
      duration: map['duration'],
      url: map['url'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'videoID': videoID,
      'sectionID': sectionID,
      'order': order,
      'title': title,
      'duration': duration,
      'url': url,
    };
  }
}
